### ISSUE TITLE
 java.lang.IllegalStateException: An instance of OnFlingListener already set.

### ISSUE THREAD

**Issue by Kolyall** [rubensousa/RecyclerViewSnap#23]
Here is my code:
```
SnapHelper snapHelper = new GravitySnapHelper(Gravity.START);
snapHelper.attachToRecyclerView(mRecyclerView);
```
but sometimes the issue occurs:
```
FATAL EXCEPTION: main
                                                                        Process: by.gramoplayer.android, PID: 12195
                                                                        java.lang.IllegalStateException: An instance of OnFlingListener already set.
                                                                            at android.support.v7.widget.SnapHelper.setupCallbacks(SnapHelper.java:114)
                                                                            at android.support.v7.widget.SnapHelper.attachToRecyclerView(SnapHelper.java:102)
                                                                            at com.github.rubensousa.gravitysnaphelper.GravitySnapHelper.attachToRecyclerView(GravitySnapHelper.java:47)
```

**Comment by evanhalley** [comment]
Set the recycler view fling listener to null (this is what worked for me).

`mRecyclerView.setOnFlingListener(null);`

**Comment by Kolyall** [comment]
I added check for isSnapHelperAttached or not

**Comment by rubensousa** [comment]
Fixed on 1.3. Now the FlingListener is set to null before the SnapHelper attaches itself to the RecyclerView.

**Comment by ghost** [comment]
getting same error with an example
```
 snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
 snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
snapAdapter.addSnap(new Snap(Gravity.CENTER_HORIZONTAL, "Snap center", apps));
```

scrolling to the bottom and then up produces this error
